language/src/regent/type_check.t

```diff
@@ -3520,6 +3520,45 @@ function type_check.expr_import_partition(cx, node)
   }
 end
 
+function type_check.expr_import_cross_product(cx, node)
+  local partitions = node.partitions:map(function(p) return type_check.expr(cx, p) end)
+  local partition_type
+  for idx, p in pairs(partitions) do
+    partition_type = std.as_read(p.expr_type)
+    if not std.is_partition(partition_type) then
+      report.error(p,
+        "type mismatch in argument " .. tostring(idx) ..
+          ": expected a partition but got " ..  tostring(partition_type))
+    end
+  end
+
+  local colors = type_check.expr(cx, node.colors)
+  local colors_type = std.as_read(colors.expr_type)
+  if colors_type ~= uint32[#partitions] then
```

Comment:
Should check with validate_implicit_cast, because (hypothetically) the user could provide a type that casts to this type.